### PR TITLE
Issue 1024 Simplify ARM Params

### LINKED ISSUE #1024: Rationalise all the ARM template parameters to simplify.
For example if both HDInsight and Avere need an SSH key combine into single param.

Also ensure all params are in param file.

### PR DESCRIPTION
**Pull request by lawrencegripper**
Closes #1024

### CODE REVIEW

**Comment by lawrencegripper** [comment]
@jjcollinge If I move HDInsight to use a key based login does this cause issues for your `init` container and can I fix them up in this PR?

**Comment by jjcollinge** [comment]
> @jjcollinge If I move HDInsight to use a key based login does this cause issues for your `init` container and can I fix them up in this PR?

@lawrencegripper it would just require a couple of changes to how the config is updated and downloaded in `setup.py`. If you want to change it over it shouldn't be too taxing or cause too much conflict.

**Comment by lawrencegripper** [comment]
Cool, I think this is all set now with the password gen removed.

File: opencga-app/app/scripts/azure/arm/solr/azuredeploy.json [hunk 1]
```diff
@@ -34,16 +34,17 @@
...
-        "description": "Admin username"
+        "description": "Admin username on all VMs."
       }
     },
-    "clusterLoginPassword": {
+    "sshKeyData": {
       "type": "securestring",
       "metadata": {
-        "description": "Admin password"
+        "description": "Admin password on all VMs."
```
Comment (marrobi) [inline opencga-app/app/scripts/azure/arm/solr/azuredeploy.json]: nit: SSH key used for all VMs

Comment (lawrencegripper) [reply opencga-app/app/scripts/azure/arm/solr/azuredeploy.json]: good spot

**Comment by lawrencegripper** [comment]
@marrobi just doing another end2end then will merge if that's ok?

**Pull request merged**